src/components/Modal.js

```diff
@@ -0,0 +1,140 @@
+import { useState, useEffect } from 'react';
+import './Modal.css';
+import { getImage } from '../helper/helps';
+import { useDispatch, useSelector } from 'react-redux'
+import { fetchPoken, initCompare, clearModal } from '../store/sliceModal';
+
+export function Modal({ visible, url, onClose }) {
+  const { left, right, is_compared, initialCompare, closeAndClean } = useManager(url, onClose);
+
+  return visible && !!left ? (
+    <div className='container'>
+      <div className='content flex column'>
+        <div className='head'>
+          <div className='flex row center'>
+            <p className='title'>{left?.name.toUpperCase()}</p>
+            {!is_compared && <button onClick={initialCompare}>CompareTo</button>}
+          </div>
+          <a className='btn' onClick={closeAndClean}>
+            X
+          </a>
+        </div>
+        { is_compared && !!right ? <Compared left={left} right={right} /> : <PokenDescription {...left} />}
+        <div>FOOTER</div>
+      </div>
+    </div>
+  ) : (
+    <></>
+  );
+}
+
+function useManager(url, onClose) {
+  const { left, right, is_compared } = useSelector(state => state.modal)
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (!!url) {
+      dispatch(fetchPoken(url))
+    }
+  }, [url]);
+
+  function initialCompare() {
+    dispatch(initCompare())
+    onClose()
+  }
+
+  function closeAndClean() {
+    dispatch(clearModal())
+    onClose()
+  }
+
+  return { initialCompare, closeAndClean, is_compared, left, right };
+}
+
+function Ul({ data }) {
+  return (
+    <ul>
+      {data?.map((element, index) => (
+        <li key={index}>{element}</li>
+      ))}
+    </ul>
+  );
+}
+
+function PokenDescription({id, desc, height, weight, gender, abilities, types}) {
```

Comment:
also naming constants or props
```suggestion
function PokenDescription({id, description, height, weight, gender, abilities, types}) {
```